RecoTracker/CkfPattern/plugins/TrajectorySegmentBuilder.cc

```diff
@@ -176,8 +175,7 @@ TrajectorySegmentBuilder::segments (const TSOS startingState)
   }
 #endif
 
-  TempTrajectoryContainer && candidates = 
-    addGroup(startingTrajectory,measGroups.begin(),measGroups.end());
+  TempTrajectoryContainer candidates = addGroup(startingTrajectory,measGroups.begin(),measGroups.end());
```

Comment:
auto may be more appropriate here.

If I understand correctly, this is guaranteed to have no cost impact only in c++17 (with required RVO).
@Dr15Jones 
Is this  on by default in everything that we build now in 104X?

If this were before c++17, should the solution be to explicitly make a copy in the return statement, while keeping this as an rvalue ref?

Reply:
regarding "using RVO before that" or even now, my question was in the context of this feature to be always enabled even with ```-O0```.  So far,  my understanding of "required" in the c++17 standard is that it will show up even with ```-O0```, compared to be optional before that.